src/ape/api/accounts.py

```diff
@@ -433,6 +466,109 @@ def get_deployment_address(self, nonce: Optional[int] = None) -> AddressType:
+        """
+        return None
+
+    def set_delegate(self, contract: ContractInstance, **txn_kwargs):
+        """
+        Have the account class override the value of it's `delegate`. For plugins that support this
+        feature, the way they choose to handle it can vary. For example, it could be a call to up-
+        grade itself using some built-in method for a smart wallet (with default txn args) e.g. the
+        Safe smart wallet (https://github.com/ApeWorX/ape-safe), or it could be to use an EIP7702-
+        like feature available on the network to set a delegate for that account. However it a
+        plugin chooses to handle it, the resulting action (if successful) should make sure that the
+        value that `self.delegate` returns is the same as `contract` after it is completed.
+
+        By default, this method raises `NotImplementedError` signaling that support is not available
+        for this feature. Calling this may result in other errors if implemented.
+
+        Args:
+            contract (`:class:~ape.contracts.ContractInstance`):
+                The contract instance to override the delegate with.
+            **txn_kwargs: Additional transaction kwargs passed to
+              :meth:`~ape.api.networks.EcosystemAPI.create_transaction`, such as ``gas``
+              ``max_fee``, or ``max_priority_fee``. For a list of available transaction
+              kwargs, see :class:`~ape.api.transactions.TransactionAPI`.
+        """
+        raise NotImplementedError
+
+    def remove_delegate(self, **txn_kwargs):
+        """
+        Has the account class remove the override for the value of it's `delegate`. For plugins
+        that support this feature, the way they choose to handle it can vary. For example, on a
+        network using an EIP7702-like feature available it will reset the delegate to empty.
+        However it a plugin chooses to handle it, the resulting action (if successful) should
+        make sure that the value that `self.delegate` returns `None` after it is completed.
+
+        By default, this method raises `NotImplementedError` signaling that support is not available
+        for this feature. Calling this may result in other errors if implemented.
+
+        Args:
+            **txn_kwargs: Additional transaction kwargs passed to
+              :meth:`~ape.api.networks.EcosystemAPI.create_transaction`, such as ``gas``
+              ``max_fee``, or ``max_priority_fee``. For a list of available transaction
+              kwargs, see :class:`~ape.api.transactions.TransactionAPI`.
+        """
+        raise NotImplementedError
+
+    @contextmanager
+    def delegate_to(
+        self,
+        contract: ContractInstance,
+        set_txn_kwargs: Optional[dict] = None,
+        remove_txn_kwargs: Optional[dict] = None,
+        **txn_kwargs,
+    ) -> Iterator[ContractInstance]:
+        """
+        Temporarily overrides the value of `delegate` for the account inside of a context manager,
```

Comment:
```suggestion
        Temporarily override the value of `delegate` for the account inside of a context manager,
```

we've been using the style of "What can I do with this method?", so verbs should be like `override` instead of `overrides`